### ISSUE TITLE
Implement HAL Specification

### ISSUE THREAD

**Issue by paulstatezny** [synapsestudios/synapse-base#49]
## Implement HAL Specification

Related resource URLs in API responses.

http://stateless.co/hal_specification.html

**Comment by baohx2000** [comment]
Returning resource links is heavily used in many rest api abstractions like OData http://www.odata.org/documentation/odata-version-4-0/

**Comment by baohx2000** [comment]
TL;DR - hal is oversimplified IMO

hal seems to contain a lot of redundant or unnecessary data pertaining to related records which it assumes the user wants.  Some of this of data should be accessible via an api metadata api call instead.  Also, it seems to want you to define any other apis for paging & searching within the result data.  This should be part of the api standardization/abstraction, not a one-off for each resource.

A good api standardization/abstraction includes methodologies for object & relation definition (metadata), paging, resource cacheing, filtering, and result data filtering (specifying what will be in the response).  Of course, that's just the request side of things.  Of course, there should also be standardization for creating/patching/deleting records, and a method for performing batch requests.
